pipeline/inferrer/feature_inferrer/app/src/batching.py

```diff
@@ -0,0 +1,119 @@
+import asyncio
+from uuid import UUID, uuid4
+from typing import Callable, Dict, Generic, List, TypeVar
+from async_timeout import timeout
+from src.logging import get_logstash_logger
+
+
+logger = get_logstash_logger(__name__)
+
+T = TypeVar("T")
+R = TypeVar("R")
+
+
+class BatchExecutionQueue(Generic[T, R]):
```

Comment:
I anticipate that this could be useful elsewhere so I documented it quite thoroughly (hence the types, I consider those part of the documentation)